### PR TITLE
[X86] Refine speed up checking clobbered FP/BP to make IPRA work.

### PR DESCRIPTION
**Pull request by FreddyLeaf**
X86 IPRA had below issue: https://gcc.godbolt.org/z/6hh88xv9r
This patch resolved it based on spillFPBP

### CODE REVIEW

File: llvm/lib/Target/X86/X86FrameLowering.cpp [hunk 1]
```diff
@@ -4567,7 +4570,9 @@ void X86FrameLowering::spillFPBP(MachineFunction &MF) const {
 
       // If the bp is clobbered by a call, we should save and restore outside of
       // the frame setup instructions.
-      if (KillMI->isCall() && DefMI != ME) {
+      // When IPRA is enabled, we could skip this step.
+      if (KillMI->isCall() && DefMI != ME && !MF.hasInlineAsm() &&
```
Comment (phoebewang) [inline llvm/lib/Target/X86/X86FrameLowering.cpp:4574]: Why checking `hasInlineAsm`?

Comment (FreddyLeaf) [reply llvm/lib/Target/X86/X86FrameLowering.cpp:4574]: Sorry, pls ignore. Will remove.

Comment (FreddyLeaf) [reply llvm/lib/Target/X86/X86FrameLowering.cpp]: [1143462](https://github.com/llvm/llvm-project/pull/109246/commits/1143462a9861c755d8db674a665c51fc35f0db4e)

**Review by phoebewang — APPROVED** [review]
LGTM.

**Comment by FreddyLeaf** [comment]
Sorry, this PR is still not correct. set as draft first.

**Comment by FreddyLeaf** [comment]
After more investigation, I realized spillFPBP has to be done before argument passing even IPRA is enabled. Considering there is a function containing e.g. 9 pointers as arguments, if there is a `push $rbp` after argument passing and before call function, the argument inside the function will not be passed correctly as it requires stack passing for the 7th, 8th, ... arguments. Closed,

**Comment by FreddyLeaf** [comment]
Created https://github.com/llvm/llvm-project/pull/109597 as a new fix for the issue here.

**Pull request closed without merging**